// react-native-checkbox-card/build/dist/RNCheckboxCard.style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports._circleCheckContainer = exports._textStyle = exports._cardStyle = void 0;
const react_native_1 = require("react-native");
const { width: ScreenWidth } = react_native_1.Dimensions.get("window");
exports._cardStyle = (height, width, borderRadius, backgroundColor) => ({
    height,
    width,
    borderRadius,
    backgroundColor,
    paddingLeft: 16,
    paddingRight: 16,
    flexDirection: "row",
    alignItems: "center",
});
exports._textStyle = (checked, checkedTextColor, uncheckedTextColor) => ({
    fontSize: 16,
    marginLeft: 16,
    fontWeight: "600",
    textDecorationLine: checked ? "line-through" : "none",
    color: checked ? checkedTextColor : uncheckedTextColor,
});
exports._circleCheckContainer = (checked, circleSize, circleBorderRadius, circleBackgroundColor, circleBorderColor) => ({
    width: circleSize,
    height: circleSize,
    borderRadius: circleBorderRadius,
    borderColor: checked ? "transparent" : circleBorderColor,
    backgroundColor: checked ? circleBackgroundColor : "transparent",
    borderWidth: 2,
    alignItems: "center",
    justifyContent: "center",
});
exports.default = react_native_1.StyleSheet.create({
    container: {
        width: ScreenWidth,
        alignItems: "center",
        justifyContent: "center",
    },
    rightIconContainer: {
        marginLeft: "auto",
    },
    sortIconImageStyle: {
        width: 10,
        height: 30,
    },
    textContainer: {
        width: "80%",
    },
    quantityTextStyle: {
        color: "#ccc",
        marginLeft: 16,
    },
    checkIconImageStyle: {
        width: 15,
        height: 15,
    },
});
